frontend-react/src/hooks/UseValueSets.ts

```diff
@@ -0,0 +1,152 @@
+import axios from "axios";
+import { useQuery } from "@tanstack/react-query";
+
+import {
+    LookupTable,
+    lookupTableApi,
+    LookupTables,
+    ValueSet,
+    ValueSetRow,
+} from "../network/api/LookupTableApi";
+
+export interface TableAttributes {
+    version: number;
+    createdAt?: string;
+    createdBy?: string;
+}
+
+/*
+
+  Fetch / Mutate functions and helpersa
+
+*/
+
+/*
+TO DISCUSS: we could set the pattern to be something like this or
+we could reference query keys within the function https://stackoverflow.com/a/68111112/5500298
+
+Have not tested, but I feel like to get the full value of react-query's refetching behavior we may want to go with the last option.
+When react-query refetches it will just re-run the function we passed to it, this would not take into account changes in variable dependencies
+unless we bake those into function itself?
+
+Note that I ran into type errors when trying to use higher order functions to avoid the inlining
+*/
+
+// will return either a value set or value set row depeding on
+// value set table name
+// we may not need the generic here, react-query probably handles that. But it couldn't hurt?
+const getLatestData = <T>(tableName: LookupTables, version: number) => {
+    const endpointHeader = lookupTableApi.getTableData<T>(version, tableName);
```

Comment:
This is nice, it seems like we have a _lot_ of flexibility in how we set up API tooling around react-query since everything is written outside of `react` now :D